cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/PreventBrokerScaleDownOperations.java

```diff
@@ -0,0 +1,156 @@
+
+                        Future<Set<String>> topicNames = topicNames(reconciliation, kafkaAdmin);
+                        descriptions = topicNames.compose(names -> {
+                            LOGGER.infoCr(reconciliation, "Topic names {}", names);
+                            return describeTopics(reconciliation, kafkaAdmin, names);
+                        });
+                        descriptions
+                                .compose(topicDescriptions -> {
+                                    LOGGER.debugCr(reconciliation, "Got {} topic descriptions", topicDescriptions.size());
+                                    Set<Integer> idsContainingPartitionReplicas = new HashSet<>();
+                                    // Provides the node IDs which the user would like to remove first
+                                    for (Integer id: idsToBeRemoved)   {
+                                        boolean result =  brokerHasAnyReplicas(reconciliation, topicDescriptions, id);
+                                        if (result) {
+                                            idsContainingPartitionReplicas.add(id);
+                                        }
+                                    }
+                                    cannotScaleDown.complete(idsContainingPartitionReplicas);
+                                    kafkaAdmin.close();
+                                    return cannotScaleDown.future();
+                                }).recover(error -> {
+                                    LOGGER.warnCr(reconciliation, "failed to get topic descriptions", error);
+                                    cannotScaleDown.fail(error);
+                                    kafkaAdmin.close();
+                                    return Future.failedFuture(error);
+                                });
+
+                    } catch (KafkaException e) {
+                        LOGGER.warnCr(reconciliation, "Kafka exception while trying to check if broker contains partition replicas", e.getMessage());
+                    }
+
+                    return resultPromise.future();
+                });
+
+        return cannotScaleDown.future();
+    }
+
+    /**
+     * This method gets the topic names after interacting with the Admin client
+     *
+     * @param reconciliation                Instance of vertx
+     * @param kafkaAdmin                    Instance of Kafka Admin
+     * @return  return the set of topic names
+     */
+    protected Future<Set<String>> topicNames(Reconciliation reconciliation, Admin kafkaAdmin) {
+        Promise<Set<String>> namesPromise = Promise.promise();
+        kafkaAdmin.listTopics(new ListTopicsOptions().listInternal(true)).names()
+                .whenComplete((names, error) -> {
+                    if (error != null) {
+                        namesPromise.fail(error);
+                    } else {
+                        LOGGER.debugCr(reconciliation, "Got {} topic names", names.size());
```

Comment:
Is this really something we want at Debug? This could be huuuuge output I think.

Reply:
We are logging the size only not the topics so it shoud just be a count?

Reply:
Sorry, missed that. That said, what it is useful for? `Got 3451 topic names` - does it really have any value? If yes, fine to keep it.

Reply:
Yeah I guess we can remove this